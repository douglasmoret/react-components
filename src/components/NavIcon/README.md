```js
const [navOpened, setNavOpened] = React.useState(false)

;<div style={{ backgroundColor: '#ebebeb', padding: 16 }}>
  <NavIcon isCloseIcon={navOpened} onClick={() => setNavOpened(!navOpened)} />
</div>
```
