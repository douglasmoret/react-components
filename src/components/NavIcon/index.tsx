import React from 'react'
import classNames from 'classnames'
import style from './style.module.scss'

interface NavIconProps {
  className?: string
  isCloseIcon: boolean
  onClick: () => void
}

const NavIcon: React.FC<NavIconProps> = ({
  className,
  isCloseIcon,
  onClick,
  ...rest
}) => {
  return (
    <div
      className={classNames(style.container, className, {
        [style.isCloseIcon]: Boolean(isCloseIcon)
      })}
    >
      <button className={style.icon} onClick={onClick} {...rest}>
        <span className={style.iconSpan} />
        <span className={style.iconSpan} />
        <span className={style.iconSpan} />
        <span className={style.iconSpan} />
      </button>
    </div>
  )
}

export default NavIcon
