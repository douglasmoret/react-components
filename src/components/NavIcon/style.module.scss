.icon {
  width: 16px;
  height: 16px;
  position: relative;
  transform: rotate(0deg);
  transition: 0.5s ease-in-out;
  cursor: pointer;
  background: transparent;
  border: none;
  outline: none;
  -webkit-tap-highlight-color: transparent;
}

.icon .iconSpan {
  display: block;
  position: absolute;
  height: 2px;
  width: 100%;
  background: var(--color-secondary-500);
  border-radius: 9px;
  opacity: 1;
  left: 0;
  transform: rotate(0deg);
  transition: 0.25s ease-in-out;
}

.isCloseIcon .iconSpan {
  background: var(--color-primary-400-contrast);
}

.icon .iconSpan:nth-child(1) {
  top: 0;
}

.icon .iconSpan:nth-child(2),
.icon .iconSpan:nth-child(3) {
  top: 6px;
}

.icon .iconSpan:nth-child(4) {
  top: 12px;
}

.isCloseIcon .iconSpan:nth-child(1) {
  top: 6px;
  width: 0%;
  left: 50%;
}

.isCloseIcon .iconSpan:nth-child(2) {
  transform: rotate(45deg);
}

.isCloseIcon .iconSpan:nth-child(3) {
  transform: rotate(-45deg);
}

.isCloseIcon .iconSpan:nth-child(4) {
  top: 6px;
  width: 0%;
  left: 50%;
}
