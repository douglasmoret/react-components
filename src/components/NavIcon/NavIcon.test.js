/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import NavIcon from './index'

test('NavIcon: simple render test', () => {
  const component = renderer.create(<NavIcon closeIcon={false} onClick={() => {}} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
