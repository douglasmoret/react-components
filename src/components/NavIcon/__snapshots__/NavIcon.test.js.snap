// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavIcon: simple render test 1`] = `
<div
  className="container"
>
  <button
    className="icon"
    closeIcon={false}
    onClick={[Function]}
  >
    <span
      className="iconSpan"
    />
    <span
      className="iconSpan"
    />
    <span
      className="iconSpan"
    />
    <span
      className="iconSpan"
    />
  </button>
</div>
`;
